taxjar/taxjar-magento2-extension
# Fix suggestion address issue

### Context
Fix some address validation issue.


### Description
- `shipping.js` - Use correct `country_id` and `region_id` keys because of the [commit changes](https://github.com/taxjar/taxjar-magento2-extension/commit/8aa43f025d4c3c26336f80696508aedb02b60d45#)
![image001](https://user-images.githubusercontent.com/3338473/120521020-16cf5b00-c39a-11eb-805f-b7fcedec2457.png)

- ~~`suggested_address_checkout_step.js` - add `waitForElement` function for `rearrangeSteps` function, because `.onestep-shipping-address` of `Aheadwork_OneStetCheckout` may render later when refresh page (ex: enter F5 key)~~ (CAN NOT Duplicate in out of box environment - reverted the changes)

- `address_validation_core.js` - add `isValidAddress` function with `address.postcode` condition, and also refactor code for nested if statements.


### Performance
None

### Testing
1. Add some product to cart and into the checkout page
2. Fill shipping address when complete with Street 1, City, Country (US), State, and Zip/Postal Code (It should enter a little incorrect for trigger address suggestion)
3. Show suggested address correctly
4. F5 (Refresh page), suggested address show correctly if an address is filled. - Aheadwork_OneStepCheckout only

#### Versions
